snapcraft/internal/parser.py

```diff
@@ -157,6 +160,8 @@ def _process_index(output):
         origin_type = data.get('origin-type')
```

Comment:
all these gets should probably raise an exception so I guess just doing project_part = data['project_part'] and the same for the others is fine. We capture the exception in whatever is looping over the documents, log a warning and continue with the next document.